### PR TITLE
Restrict `++` and `--` to slack IDs

### PR DESCRIPTION
**Pull request by aseem09**

### CODE REVIEW

File: scripts/leaderboard.coffee [hunk 1]
```diff
@@ -116,23 +117,31 @@ module.exports = (robot) ->
...
+        # generate response message for reply
+        if result.Response == "-1"
+          newmsg = "#{testword} [Sorry, You can't give ++ or -- to yourself.]"
+        else
+          newmsg = "#{testword} [#{result.Response} #{result.Name} now at #{result.New}] "
+        oldmsg = oldmsg.substr(0, start) + newmsg + oldmsg.substr(end+1)
+        start += newmsg.length
+    
+    # reply with updated message, if modified
+    if oldmsg != msg.message.text
```
Comment (aseem09) [inline scripts/leaderboard.coffee:144]: Why have you added this check?

Comment (BURG3R5) [reply scripts/leaderboard.coffee:144]: [This loop](https://github.com/BURG3R5/bot/blob/master/scripts/leaderboard.coffee#L118) iterates over every `++`/`--`. [This if](https://github.com/BURG3R5/bot/blob/master/scripts/leaderboard.coffee#L121) checks if the `++`/`--` is valid. The means only the first `++` in "@bert++ for mastering C++" will be operated on and modified in the reply message. Now, if _all_ the `++`/`--` in the message are invalid, like "C++ is a nightmare!", [this if](https://github.com/BURG3R5/bot/blob/master/scripts/leaderboard.coffee#L143) prevents an unmodified message from being sent.

In other words, if the message is not changed by Bro, don't send it.

**Comment by BURG3R5** [comment]
[This function](https://github.com/BURG3R5/bot/blob/fd5b20bfad029b76d93d0b271aec145c9306e8d5/scripts/leaderboard.coffee#L184) is inefficient. It's copied from `info.coffee`, so it searches for multiple matches, which can take time. Since we already know there's gonna be a single match, we should return the first match in the `for` loop. I don't have enough knowledge of Coffeescript to know how to do that.

**Comment by BURG3R5** [comment]
Tried to fix search inefficiency in 759541ade837629ac21586e3c561fb629279f1ad. Kindly review and advise.

File: scripts/leaderboard.coffee (hunk 1 above)
Comment (aseem09) [reply scripts/leaderboard.coffee:144]: Cool
